### ISSUE TITLE
Error when displaying results outside REPL

### ISSUE THREAD

**Issue by eddieschoute** [piever/GroupedErrors.jl#9]
I am trying to plot some results using a simple terminal execution (e.g. `julia Main.jl`) instead of the REPL. I have the following plotting code:
```
plotResult = @> dataDf begin
	@splitby _.markedItems
	@x _.XAxis
	@across _.iteration
	@y _.results (mean, sem)
	@plot plot(show=true)
end
```

This works fine inside the REPL and opens a browser with the plot. However, when run directly I get the following error:
```
ERROR: LoadError: MethodError: no method matching json(::Array{Any,1})
The applicable method may be too new: running in world age 24366, while current world is 24367.
Closest candidates are:
  json(::Any) at /Users/eddie/.julia/v0.6/JSON/src/Writer.jl:340 (method too new to be called from this world context.)
  json(::Any, !Matched::Any) at /Users/eddie/.julia/v0.6/JSON/src/Writer.jl:341 (method too new to be called from this world context.)
Stacktrace:
 [1] plotly_series_json(::Plots.Plot{Plots.PlotlyBackend}) at /Users/eddie/.julia/v0.6/Plots/src/backends/plotly.jl:705
 [2] html_body(::Plots.Plot{Plots.PlotlyBackend}, ::Void) at /Users/eddie/.julia/v0.6/Plots/src/backends/plotly.jl:725
 [3] #standalone_html#263(::String, ::Function, ::Plots.Plot{Plots.PlotlyBackend}) at /Users/eddie/.julia/v0.6/Plots/src/backends/web.jl:8
 [4] (::Plots.#kw##standalone_html)(::Array{Any,1}, ::Plots.#standalone_html, ::Plots.Plot{Plots.PlotlyBackend}) at ./<missing>:0
 [5] write_temp_html(::Plots.Plot{Plots.PlotlyBackend}) at /Users/eddie/.julia/v0.6/Plots/src/backends/web.jl:36
 [6] standalone_html_window at /Users/eddie/.julia/v0.6/Plots/src/backends/web.jl:45 [inlined]
 [7] _display at /Users/eddie/.julia/v0.6/Plots/src/backends/plotly.jl:756 [inlined]
 [8] display(::Plots.PlotsDisplay, ::Plots.Plot{Plots.PlotlyBackend}) at /Users/eddie/.julia/v0.6/Plots/src/output.jl:145
 [9] _do_plot_show(::Plots.Plot{Plots.PlotlyBackend}, ::Bool) at /Users/eddie/.julia/v0.6/Plots/src/output.jl:152
 [10] _plot!(::Plots.Plot{Plots.PlotlyBackend}, ::Dict{Symbol,Any}, ::Tuple{Array{Float64,1},Array{Float32,1}}) at /Users/eddie/.julia/v0.6/Plots/src/plot.jl:242
 [11] #plot#145(::Array{Any,1}, ::Function, ::Array{Float64,1}, ::Vararg{Any,N} where N) at /Users/eddie/.julia/v0.6/Plots/src/plot.jl:56
 [12] (::RecipesBase.#kw##plot)(::Array{Any,1}, ::RecipesBase.#plot, ::Array{Float64,1}, ::Array{Float32,1}, ::Vararg{Array{Float32,1},N} where N) at ./<missing>:0
 [13] #plot_helper#42(::Array{Any,1}, ::Function, ::GroupedErrors.ProcessedTable{IndexedTables.IndexedTable{Tuple{Float64,Tuple{Float32,Float32}},Tuple{Int64,Int64},IndexedTables.Columns{Tuple{Int64,Int64},Tuple{Array{Int64,1},Array{Int64,1}}},IndexedTables.Columns{Tuple{Float64,Tuple{Float32,Float32}},Tuple{Array{Float64,1},Array{Tuple{Float32,Float32},1}}}}}, ::Function, ::Symbol) at /Users/eddie/.julia/v0.6/GroupedErrors/src/plot_macro.jl:32
 [14] (::GroupedErrors.#kw##plot_helper)(::Array{Any,1}, ::GroupedErrors.#plot_helper, ::GroupedErrors.ProcessedTable{IndexedTables.IndexedTable{Tuple{Float64,Tuple{Float32,Float32}},Tuple{Int64,Int64},IndexedTables.Columns{Tuple{Int64,Int64},Tuple{Array{Int64,1},Array{Int64,1}}},IndexedTables.Columns{Tuple{Float64,Tuple{Float32,Float32}},Tuple{Array{Float64,1},Array{Tuple{Float32,Float32},1}}}}}, ::Function, ::Symbol) at ./<missing>:0
 [15] #plot_helper#53(::Array{Any,1}, ::Function, ::GroupedErrors.Selector{DataFrames.DataFrame,Plotting.##2#7,GroupedErrors.##2#7,Plotting.##4#9,Plotting.##3#8,Plotting.##5#10}, ::Function, ::Symbol) at /Users/eddie/.julia/v0.6/GroupedErrors/src/plot_macro.jl:42
 [16] (::GroupedErrors.#kw##plot_helper)(::Array{Any,1}, ::GroupedErrors.#plot_helper, ::GroupedErrors.Selector{DataFrames.DataFrame,Plotting.##2#7,GroupedErrors.##2#7,Plotting.##4#9,Plotting.##3#8,Plotting.##5#10}, ::Function, ::Symbol) at ./<missing>:0
 [17] plotData(::IndexedTables.IndexedTable{NamedTuples._NT_results{Float32},Tuple{Int64,Int64,Int64},IndexedTables.Columns{NamedTuples._NT_markedItems_repeat_iteration{Int64,Int64,Int64},NamedTuples._NT_markedItems_repeat_iteration{Array{Int64,1},Array{Int64,1},Array{Int64,1}}},IndexedTables.Columns{NamedTuples._NT_results{Float32},NamedTuples._NT_results{Array{Float32,1}}}}, ::Int64) at /Users/eddie/dev/single-qubit/Plotting.jl:17
 [18] main() at /Users/eddie/dev/single-qubit/Main.jl:51
 [19] include_from_node1(::String) at ./loading.jl:576
 [20] include(::String) at ./sysimg.jl:14
 [21] process_options(::Base.JLOptions) at ./client.jl:305
 [22] _start() at ./client.jl:371
while loading /Users/eddie/dev/single-qubit/Main.jl, in expression starting on line 58
```

The behavior in the REPL is different from the command line. Any ideas on how to get the html output, or at least some output (preferably pdf/eps)? For example, `Plots.savefig(plotResult, "plot.pdf")` also gives a warning:
```
WARNING: _show is not defined for this backend. m=application/pdf
```
and results in an empty `plot.pdf` file.

**Comment by piever** [comment]
I think the issue is mostly unrelated with this package.

With order:

- using the REPL (or some IDE such as Jupyter Notebook, Juno or JuliaVS Code) is much better than running the script with `julia Main.jl`, mainly because by running a script with `julia Main.jl` you incur everytime in the "precompilation cost", that is to say the program takes very long because it's being precompiled (which for Plots.jl takes a while...). If you do it from the REPL you'll see that the second time you run the code, it's much faster.

- the specific issue may be with Plots.jl or `plotly`: I'd recommend trying to just do some plot command from Plots.jl that do not involve this package. For example, if Main.jl only contains:

```julia
using Plots
plot(rand(10))
```
do you still get the error?

- the Plots package can use many different backends to draw the plot (see [here](http://docs.juliaplots.org/latest/backends/) for details). If none is installed, it will default to the online Plot.ly plotting service, from which I believe you can only save to html with

`Plots.savefig(plotResult, "plot.html")`

You can of course later export the html to pdf from your browser but that's not ideal. You should probably install a backend (such as [PlotlyJS](https://github.com/sglyon/PlotlyJS.jl), which is the same thing but downloaded to your computer rather than online, I use [GR](https://github.com/jheinen/GR.jl), [PyPlot](https://github.com/JuliaPy/PyPlot.jl) is also an option) which should allow to save it in vectorial format. After doing `Pkg.add("GR")` at the REPL, something like:

```julia
using Plots
gr() #select plotting backend, plotlyjs() or pyplot() are also possible if you have the packages installed
plotResult = @> dataDf begin
	@splitby _.markedItems
	@x _.XAxis
	@across _.iteration
	@y _.results (mean, sem)
	@plot plot(show=true)
end
Plots.savefig(plotResult, "plot.pdf")
```
should work. `svg` is also supported (it infers the format from the extension, just write `Plots.savefig(plotResult, "plot.svg")`), `eps` I'm not so sure.

All of these is probably explained much more clearly in the [Plots.jl documentation](http://docs.juliaplots.org/latest/).

**Comment by eddieschoute** [comment]
1. I was initially not aware of how inefficient Julia is when loading packages. Also I was planning to run the code on a remote computer so a REPL would not make much sense in that case (the startup overhead is negligible there anyway).
2. The given code works, but if I add `show=true` it fails with a similar error message.
3. I thought I had tried, but in a simple example the savefig method does work for html. It doesn't work for my case unfortunately. And I tried using the GR backend to create a pdf file but no file is created for some reason No errors/warnings are shown either.
